options/options.js

```diff
@@ -16,21 +16,87 @@ function saveOptions(e) {
 function restoreOptions() {
   var value = '';
   var getting = '';
-  document.querySelectorAll('input[id]').forEach(function(item) {
-    item = item.id;
+  document.querySelectorAll('input[id]').forEach(function(el) {
+    var item = el.id;
     getting = browser.storage.local.get(item);
     getting.then(function(result) {
       value = result[Object.keys(result)[0]];
-      if (Number.isInteger(parseInt(value)) || document.querySelector('#' + item).type === 'url') {
-		if(value !== undefined) {
-		  document.querySelector('#' + item).value = value;
-		}
+      var isInt = Number.isInteger(parseInt(value, 10));
```

Comment:
Always pass `10` as second argument to `parseInt`. Otherwise, funny things will happen if the first argument starts with `0` ;-)